src/transforms/code-scanning-alert.ts

```diff
@@ -1,35 +1,80 @@
 import issueKeyParser from "jira-issue-key-parser";
-import { Context } from "probot/lib/context";
-import { JiraRemoteLinkData } from "../interfaces/jira";
+import {Context} from "probot/lib/context";
+import {JiraRemoteLinkData} from "../interfaces/jira";
 
-function getLastUpdated({ status, started_at, completed_at }) {
-	if (status === 'completed') {
-		return completed_at;
-	} else {
-		return started_at;
+// TODO ARC-618
+const getPullRequestTitle = (id: number): string => {
+	return `GH-${id}`;
+}
+
+const getEntityTitle = (ref: string): string => {
+	// ref can either be a branch reference or a PR reference
+	const components = ref.split("/")
+	switch (components[1]) {
+		case "heads": // branch
+			// The branch name may contain forward slashes! Rejoin them
+			return components.slice(2).join("/");
+		case "pull": // pull request
+			return getPullRequestTitle(parseInt(components[2]));
+		default:
+			// log something here
+			return "";
+	}
+}
+
+const transformStatusToAppearance = (status: string): string => {
+	switch (status) {
+		case "open":
+			return "removed"; // red
+		case "fixed":
+			return "success"; // green
+		case "dismissed":
+			return "moved"; // yellow
+		default:
+			return "default";
 	}
 }
 
 export default (context: Context): JiraRemoteLinkData => {
-	const { code_scanning_alert } = context.payload;
-	const issueKeys = issueKeyParser().parse(`${code_scanning_alert.instances.ref}`);
+	const { action, alert, ref, repository} = context.payload;
+
+	// Grab branch names or PR titles
+	const entityTitles = [];
+	if (action === "closed_by_user" || action === "reopened_by_user") {
+		// These are manual operations done by users and are not associated to a specific Issue.
+		// The webhook contains ALL instances of this alert, so we need to grab the ref from each instance.
+		entityTitles.push(...alert.instances.map((instance) => getEntityTitle(instance.ref)));
+	} else {
+		// The action is associated with a single branch/PR
+		entityTitles.push(getEntityTitle(ref));
+	}
+
+	const parser = issueKeyParser();
+	// Flatten and get all issue keys
+	const issueKeys = [].concat(...entityTitles.map((entityTitle) => parser.parse(entityTitle)));
```

Comment:
You can just do `issueKeyParser().parse(entityTitle)`

Reply:
Won't this create a whole new object every iteration through `entityTitles` ?